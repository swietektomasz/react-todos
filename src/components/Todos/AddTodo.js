import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { addTodo } from '../../store/todos/actions'

export const AddTodo = props => {
  const [title, setTitle] = useState('')
  return (
    <div>
      <input
        placeholder="add todo..."
        defaultValue=""
        onChange={({ target: { value } }) => setTitle(value)}
      />
      <button onClick={() => props.addTodo(title)}>add</button>
    </div>
  )
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      addTodo,
    },
    dispatch,
  )
}

export default connect(
  null,
  mapDispatchToProps,
)(AddTodo)

AddTodo.propTypes = {
  addTodo: PropTypes.func.isRequired,
}
