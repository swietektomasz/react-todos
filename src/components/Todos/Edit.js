import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { Formik, Field } from 'formik'

import { TextInput } from '../../Form/Input'
import { updateTodoWithDispatch, getTodoByIdWithDispatch } from '../../store/todos'

class Edit extends Component {
  componentDidMount = () => {
    this.props.getTodoByIdWithDispatch(this.props.match.params.id)
  }

  render() {
    const {
      byId: { description, title, done, id },
    } = this.props
    const initialValues = {
      description,
      title,
      done,
      id,
    }
    return (
      <Formik
        enableReinitialize
        initialValues={initialValues}
        onSubmit={(values, actions) => {
          this.props.updateTodoWithDispatch(values)
          actions.setSubmitting(false)
          this.props.history.push(`/todos/`)
        }}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <Field name="title" component={TextInput} label={'Title'} />
            <Field name="description" component={TextInput} label={'Description'} />
            <input type="hidden" name="id" />
            <button type="submit">edit</button>
          </form>
        )}
      ></Formik>
    )
  }
}

const mapStateToProps = ({ todos }) => {
  return todos
}

const mapDispatchToProps = {
  updateTodoWithDispatch,
  getTodoByIdWithDispatch,
}

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(Edit),
)
