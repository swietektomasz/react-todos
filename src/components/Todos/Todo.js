import React from 'react'
import PropTypes from 'prop-types'

export const Todo = ({ handleChecking, title, done, id }) => (
  <div>
    <label>{title}</label>
    <input type="checkbox" onChange={() => handleChecking(id)} defaultChecked={done} />
  </div>
)

export default Todo

Todo.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  done: PropTypes.bool,
  handleChecking: PropTypes.func,
}
