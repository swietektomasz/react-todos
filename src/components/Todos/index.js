import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Todo from './Todo'

export default class TodoList extends Component {
  static propTypes = {
    list: PropTypes.array,
    id: PropTypes.number,
  }

  render() {
    const { list = [] } = this.props

    return (
      <div>
        {list.map(todo => (
          <Todo
            key={todo.id}
            title={todo.title}
            done={todo.done}
            id={todo.id}
            handleChecking={() => this.props.updateCheckedTodos(this.props.id)}
          />
        ))}
      </div>
    )
  }
}
