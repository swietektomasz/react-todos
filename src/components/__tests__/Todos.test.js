import React from 'react'
import renderer from 'react-test-renderer'

import Todos from '../Todos'

jest.mock('../Todos/Todo', () => () => <div>Todo</div>)

describe('<Todos />', () => {
  const defaultProps = {}
  const wrapper = renderer.create(<Todos {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('renders one item', () => {
    const defaultProps = {
      list: [{ id: 0, title: 'test', done: false }],
      updateCheckedTodos: jest.fn(),
    }
    const wrapper = renderer.create(<Todos {...defaultProps} />)
    const instance = wrapper.root

    expect(instance.findByType('div')).toBeDefined()
  })

  test('renders three items', () => {
    const defaultProps = {
      list: [
        { id: 0, title: 'test', done: false },
        { id: 1, title: 'test', done: false },
        { id: 2, title: 'test', done: false },
      ],
      updateCheckedTodos: jest.fn(),
    }
    const wrapper = renderer.create(<Todos {...defaultProps} />)
    const instance = wrapper.root

    expect(instance.findByType('div')).toBeDefined()
    expect(instance.findAllByType('div')).toHaveLength(4)
    const jsonInstance = wrapper.toJSON()
    expect(jsonInstance.children).toHaveLength(3)
  })
})
