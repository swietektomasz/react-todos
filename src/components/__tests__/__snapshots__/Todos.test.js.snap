exports[`<Todos /> render 1`] = `<div />`;
